InternxtDesktop/Services/BackupsService.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  BackupsService.swift
+//  InternxtDesktop
+//
+//  Created by Richard Ascanio on 1/8/24.
+//
+
+import Foundation
+import RealmSwift
+
+class BackupsService: ObservableObject {
+    @Published var foldernames: [FoldernameToBackup] = []
+    @Published var urls: [URL] = []
+    @Published var isBackupButtonEnabled: Bool = false
+
+    private func getRealm() -> Realm {
+        do {
+            return try Realm(fileURL: ActivityManager.realmURL)
+        } catch {
+            error.reportToSentry()
+            fatalError("Unable to open Realm")
+        }
+
+    }
+
+    func clean() throws {
+        foldernames = []
+        urls = []
+        let realm = getRealm()
+        try realm.write {
+            realm.deleteAll()
+        }
+    }
+
+    func addFoldernameToBackup(_ foldernameToBackup: FoldernameToBackup) {
+        guard let url = URL(string: foldernameToBackup.url) else {
+            return
+        }
+
+        do {
+            let realm = getRealm()
+            try realm.write {
+                realm.add(foldernameToBackup)
+            }
+            self.urls.append(url)
+        } catch {
+            error.reportToSentry()
```

Comment:
errors should be handled by the consumer view, always bubble up errors, and report it to Sentry at the end, if this fails, the only one that will know this will be Sentry, in the view we won't have any way to know if this failed as the catch will silence it